apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis.erl

```diff
@@ -30,8 +33,26 @@ roots() ->
 
 fields("config_producer") ->
     emqx_bridge_schema:common_bridge_fields() ++
-        emqx_resource_schema:fields("resource_opts") ++
-        fields(connector_config) ++ fields(producer);
+        fields("resource_opts") ++
+        fields(connector_config) ++
+        fields(producer);
+fields("resource_opts") ->
+    [
+        {resource_opts,
+            mk(
+                ref(?MODULE, "creation_opts"),
+                #{required => true, desc => ?DESC(emqx_resource_schema, "creation_opts")}
+            )}
+    ];
+fields("creation_opts") ->
+    emqx_resource_schema:create_opts([
+        {batch_size, #{
+            validator => emqx_resource_validator:max(int, 500)
+        }},
+        {max_buffer_bytes, #{
```

Comment:
I think we shouldn't limit this, as it doesn't directly control the size of batches.

If one configures the batch size to one and ensures that each message has acceptable size, there's no reason to prevent accumulation of more than 5 MB in the buffer.

Reply:
ACK